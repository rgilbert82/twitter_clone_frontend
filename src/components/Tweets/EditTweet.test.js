jest.mock('../../services/aws');

import React from 'react';
import { shallow } from 'enzyme';
import { EditTweet } from '.';

describe(EditTweet, () => {
  const currentUser = { id: 1, name: 'Ryan', username: 'RG', token: '12345', slug: 'rg', followers: 1, following: 1, tweet_count: 1, retweet_count: 1, all_tweet_count: 2 };
  const editMock = jest.spyOn(EditTweet.prototype, 'uploadPhotoBeforeCreatingTweet');
  const deleteMock = jest.fn();
  const tweetBody = 'Hello World';
  let component;

  beforeEach(() => {
    component = shallow(<EditTweet body={tweetBody} editTweetBody={editMock} deleteTweet={deleteMock} displayMessage={jest.fn()} currentUser={currentUser} />);
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("toggles the form", () => {
    expect(component.state('editable')).toEqual(false);
    component.find('.formToggle').simulate('click');
    expect(component.state('editable')).toEqual(true);
    component.find('.formToggle').simulate('click');
    expect(component.state('editable')).toEqual(false);
  });

  it("edits the body", () => {
    expect(component.state('editBody')).toEqual(tweetBody);
    component.find('.formToggle').simulate('click');
    component.find('#form_body').simulate('change', { target: { value: 'Hi' }});
    expect(component.state('editBody')).toEqual('Hi');
  });

  it("submits the form", () => {
    component.find('.formToggle').simulate('click');
    component.find('#form_body').simulate('change', { target: { value: 'Hi' }});
    component.find('#submitForm').simulate('click');
    expect(editMock).toBeCalled();
  });
});
