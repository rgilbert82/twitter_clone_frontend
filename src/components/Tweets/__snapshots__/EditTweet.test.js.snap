// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class EditTweet extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = {
      editBody: this.props.body,
      editable: false };


    this.toggleForm = this.toggleForm.bind(this);
    this.updateTweet = this.updateTweet.bind(this);
    this.editTweet = this.editTweet.bind(this);
    this.validForm = this.validForm.bind(this);
    this.deleteTweet = this.deleteTweet.bind(this);
    this.uploadPhotoBeforeCreatingTweet = this.uploadPhotoBeforeCreatingTweet.bind(this);
  }

  toggleForm() {
    this.setState({
      editBody: this.props.body,
      editable: !this.state.editable });

  }

  updateTweet(e) {
    this.setState({ editBody: e.target.value });
  }

  validForm() {
    return this.state.editBody.length > 0;
  }

  uploadPhotoBeforeCreatingTweet() {
    return (0, _aws.uploadPhoto)(this.props.currentUser.id).
    then(photoPath => {
      this.editTweet(photoPath);
    }).catch(err => {
      this.props.displayMessage('There was an error uploading your image');
    });
  }

  editTweet(photoPath) {
    this.setState({ editable: false });
    this.props.editTweetBody({ body: this.state.editBody, image: photoPath });
  }

  deleteTweet() {
    if (window.confirm('Are you sure?')) {
      this.props.deleteTweet();
    }
  }

  render() {
    let component;

    if (this.state.editable) {
      component =
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 61 }, __self: this },
        _react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 62 }, __self: this },
          _react2.default.createElement(_reactBootstrap.FormControl, { componentClass: 'textarea', id: 'form_body', maxLength: '140', onChange: this.updateTweet, value: this.state.editBody, placeholder: 'Edit your tweet', __source: { fileName: _jsxFileName, lineNumber: 63 }, __self: this })),

        _react2.default.createElement(_reactBootstrap.FormGroup, { className: 'photo_upload_modal_wrapper', __source: { fileName: _jsxFileName, lineNumber: 65 }, __self: this },
          _react2.default.createElement(_reactBootstrap.ControlLabel, { htmlFor: 'photo_upload', __source: { fileName: _jsxFileName, lineNumber: 66 }, __self: this }, 'Change your image'),
          _react2.default.createElement(_reactBootstrap.FormControl, { type: 'file', id: 'photo_upload', className: 'photo_upload_modal', __source: { fileName: _jsxFileName, lineNumber: 67 }, __self: this })),

        _react2.default.createElement('div', { id: 'edit_tweet_buttons', __source: { fileName: _jsxFileName, lineNumber: 69 }, __self: this },
          _react2.default.createElement(_reactBootstrap.Button, { id: 'submitForm', onClick: this.uploadPhotoBeforeCreatingTweet, disabled: !this.validForm(), __source: { fileName: _jsxFileName, lineNumber: 70 }, __self: this }, 'Submit'),
          _react2.default.createElement(_reactBootstrap.Button, { className: 'formToggle', onClick: this.toggleForm, __source: { fileName: _jsxFileName, lineNumber: 71 }, __self: this }, 'Cancel'),
          _react2.default.createElement(_reactBootstrap.Button, { id: 'deleteTweet', onClick: this.deleteTweet, __source: { fileName: _jsxFileName, lineNumber: 72 }, __self: this }, 'Delete')));



    } else {
      component =
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 78 }, __self: this },
        _react2.default.createElement(_reactBootstrap.Button, { className: 'formToggle', onClick: this.toggleForm, __source: { fileName: _jsxFileName, lineNumber: 79 }, __self: this }, 'Edit'));


    }

    return component;
  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditTweet
    body="Hello World"
    currentUser={
        Object {
            "all_tweet_count": 2,
            "followers": 1,
            "following": 1,
            "id": 1,
            "name": "Ryan",
            "retweet_count": 1,
            "slug": "rg",
            "token": "12345",
            "tweet_count": 1,
            "username": "RG",
          }
    }
    deleteTweet={[Function]}
    displayMessage={[Function]}
    editTweetBody={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="formToggle"
        disabled={false}
        onClick={[Function]}
>
        Edit
</Button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "block": false,
        "bsClass": "btn",
        "bsStyle": "default",
        "children": "Edit",
        "className": "formToggle",
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Edit",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="formToggle"
          disabled={false}
          onClick={[Function]}
>
          Edit
</Button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Edit",
          "className": "formToggle",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
